### ISSUE TITLE
Unable to turn off `Missing type annotation` swiggles

### ISSUE THREAD

**Issue by choonkeat** [elm-tooling/elm-language-server#639]
For an elm source file [like this](https://github.com/choonkeat/demo-bug-example/blob/49f3e547aa51a866efd6601abcc28f8bd7d2983c/src/Main.elm)

```elm
module Main exposing (..)


hello =
    "world"
```

there's a warning swiggle line below `hello` that cannot be turned off, even if `Disable Elm LSDiagnostics` was set in vscode settings.json `"elmLS.disableElmLSDiagnostics": true`

## Expected Behavior

I was hoping either setting vscode settings.json `"elmLS.disableElmLSDiagnostics": true` OR setting `elm-analyse.json` [`"MissingTypeAnnotation": false`](https://github.com/elm-tooling/elm-language-client-vscode/blob/5aa074d8cd40617e9643215b8dd46cd5676ec4df/schemas/elm-analyse.schema.json#L61-L63) should have made the warning swiggle lines go away

## Current Behavior

The `Missing type annotation` swiggle continue to be present. In the screenshot below, I'm debugging from this main branch at a522ee635f6edf519fdf6e668ee028711aa66d9f

<img width="1723" alt="Screenshot 2021-09-18 at 3 32 42 PM" src="https://user-images.githubusercontent.com/473/133880641-d922e5fb-ebcf-40a1-aadb-b76a0c868e16.png">

## Steps to Reproduce (for bugs)

1. Git clone https://github.com/choonkeat/demo-bug-example
2. Open in vscode

## Context

For regular functions, we write type annotations. In some files, there are top level records that are more like a bunch of config values that we use in the same file

```elm
view : State -> Html Msg
view state =
    div [ css styles.container ]
        [ div [ css styles.infoContainer ]
            [ div [ css styles.infoSubTitle ] [ text "this is an example" ]

            -- elided ...
            ]
        ]


styles =
    { container =
        [ Tw.space_y_3
        , Bp.sm
            [ Tw.space_y_6
            ]
        ]
    , infoContainer =
        [ Tw.text_center
        , Tw.font_bold
        , Tw.text_gray_500
        , Bp.sm
            [ Tw.my_10
            ]
        ]
    , infoSubTitle =
        [ Tw.text_lg
        , Bp.sm
            [ Tw.text_xl
            ]
        ]
    , infoMain =
        [ Tw.text_3xl
        ]
    , hideWhenMd =
        [ Bp.md [ Tw.hidden ]
        ]
    , infoFootNote =
        [ Tw.text_sm
        , Tw.font_normal
        , Tw.mt_2
        ]
    , buttonContainer =
        [ Tw.pt_6
        , Tw.flex
        , Tw.justify_end
        ]
    , button =
        [ Tw.px_6
        , Tw.py_2
        , Tw.rounded_lg
        , Tw.text_base
        , Tw.text_white
        , Css.hover
            [ Tw.bg_yellow_600
            ]
        ]
    }
```

that having a type annotation is not useful and clutters the source code. 

```elm
styles : { container : List Css.Style, infoContainer : List Css.Style, infoSubTitle : List Css.Style, infoMain : List Css.Style, hideWhenMd : List Css.Style, infoFootNote : List Css.Style, buttonContainer : List Css.Style, button : List Css.Style }
```

as view code and config needs evolve, we'd need to remove the type annotation (because it's wrong now) then click to add the new suggested annotation 😕

_NOTE: It's probably a distraction to discuss if this is good practice, but the question here remains: can this be turned off (a bug) or it cannot be (no such feature)? 🙇_

## Your Environment
* Version used: regular vscode v2.2.1 but also tried in a522ee635f6edf519fdf6e668ee028711aa66d9f
* Editor name and version (e.g. VSCode 1.36.1): VScode `Version: 1.56.2 Commit: 054a9295330880ed74ceaedda236253b4f39a335`
* Environment name and version (e.g. node.js 5.4): nodejs `v14.17.0`
* Operating System and version: MacOS 11.6 (20G165)

**Comment by razzeee** [comment]
Hey,

your `elm-analyse.json` looks wrong to me. can yo try again with:

```
{
  "checks": { "MissingTypeAnnotation": false }
}
```

**Comment by razzeee** [comment]
On disabling this via `disableElmLSDiagnostics`, this warning seems a bit different to me, as it's a suggestion. So it might make sense to have another setting for this if we want to do it.

**Comment by choonkeat** [comment]
> your elm-analyse.json looks wrong to me. can yo try again with

though not for v2.2.1, that fixed it for a522ee6 perfect! thanks

> as it's a suggestion. So it might make sense to have another setting for this if we want to do it.

true. maybe `disableElmAnalyse` if wish to proceed. in the meantime. I'll have that `elm-analyse.json` for when a522ee6 is released 👍
